runtime/vm/ValueTypeHelpers.hpp

```diff
@@ -417,6 +417,81 @@ class VM_ValueTypeHelpers {
 		}
 	}
 
+	/**
+	 * Performs an aastore operation on an object. Handles flattened and non-flattened cases.
+	 *
+	 * Assumes recieverObject and paramObject are not null.
```

Comment:
Should I add that it assumes the index is less than the array size and that it does not throw exceptions?

Reply:
yes

Reply:
you can add that all AIOB exceptions must be thrown before hand